pkg/etcd/etcd_test.go

```diff
@@ -0,0 +1,198 @@
+package etcd
+
+import (
+	"context"
+	"fmt"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	"go.uber.org/zap/zapcore"
+
+	corev1 "k8s.io/api/core/v1"
+	policyv1 "k8s.io/api/policy/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/client/fake"
+	zaplog "sigs.k8s.io/controller-runtime/pkg/log/zap"
+
+	commonLabels "github.com/medik8s/common/pkg/labels"
+)
+
+const (
+	etcdQuorumPDBName = "etcd-guard-pdb" // The new name of the PDB - From OCP 4.11
+)
+
+var guardLabel = map[string]string{"app": "guard"}
+
+var _ = Describe("Check if etcd disruption is allowed", func() {
+	opts := zaplog.Options{
+		Development: true,
+		TimeEncoder: zapcore.RFC3339NanoTimeEncoder,
+	}
+	log := zaplog.New(zaplog.WriteTo(GinkgoWriter), zaplog.UseFlagOptions(&opts))
```

Comment:
This should go into BeforeSuite.
Actually, we already have something similar over there...?

Reply:
Are you referring to the log or the fake controller?

Reply:
log

Reply:
To create a log per test? Isn't one log would be enough for all the unit tests?

Reply:
one log is enough, yes, and that's why it can or even should go into BeforeSuite, where we already have some log setup, which is duplicated here